crates/prql-parser/src/lexer.rs

```diff
@@ -140,6 +127,38 @@ pub fn ident_part() -> impl Parser<char, String, Error = Cheap<char>> {
     plain.or(backticks)
 }
 
+fn new_line() -> impl Parser<char, Token, Error = Cheap<char>> {
+    just('\n').to(Token::NewLine)
+}
+
+fn whitespace() -> impl Parser<char, (), Error = Cheap<char>> {
+    one_of("\t \r").repeated().at_least(1).ignored()
+}
+
+fn comment() -> impl Parser<char, (), Error = Cheap<char>> {
+    let comment = just('#').then(none_of('\n').repeated());
+
+    comment
+        .separated_by(new_line().then(whitespace().or_not()))
+        .at_least(1)
+        .ignored()
+}
+
+fn ignored() -> impl Parser<char, (), Error = Cheap<char>> {
+    comment()
+        .or(whitespace())
+        .or(line_continuation())
+        .repeated()
+        .ignored()
+}
+
+fn line_continuation() -> impl Parser<char, (), Error = Cheap<char>> {
+    just('\n')
+        .then(whitespace().repeated().or_not())
+        .then(just('\\'))
+        .ignored()
+}
```

Comment:
I split some parsers out to test the new function individually; on reflection maybe not worthwhile, am happy to lump them all back into the big func if folks prefer that

Reply:
I put them back, which speeds up compilation (not just parsing or even lexing) by 1.5%...